### PR TITLE
Fix netcore WebClient.UploadValuesAsync() UrlEncode behavior

### LINKED ISSUE #18674: WebClient.UploadValuesAsync app-compat difference with Desktop
While investigating #17882, I root caused the failure of the WebClient.UploadValuesAsync test.

On .NET Framework, WebClient uses an internal method, `UrlEncodeBytesToBytesInternal`, to "UrlEncode" the NameValueCollection. This is causing it to use lowercase hex digits.

On .NET Core, WebClient uses the public WebUtility.UrlEncode method which always uses uppercase hex digits.

The public WebUtility.UrlEncode is behaving the same (uppercase digits) on both .NET Framework and .NET Core.

However, .NET Core WebClient is sending request body data using uppercase hex digits while on .NET Framework it is using lowercase hex digits while sending the request body data.

For app-compat, we should change .NET Core WebClient's implementation to match .NET Framework.

### PR DESCRIPTION
**Pull request by schen381**
Change request body data of .Net Core WebClient from using uppercase hex digits to using lowercase hex digits.

Fix #18674

### CODE REVIEW

**Comment by schen381** [comment]
@dotnet-bot test Outerloop Windows_NT Debug please
@dotnet-bot test Outerloop Ubuntu14.04 Debug please
@dotnet-bot test Outerloop OSX10.12 Debug please

File: src/System.Net.WebClient/src/System/Net/WebClient.cs [hunk 1]
```diff
@@ -1176,6 +1176,101 @@ private string MapToDefaultMethod(Uri address)
...
+            byte[] bytes = Encoding.UTF8.GetBytes(str);
+            return Encoding.ASCII.GetString(UrlEncodeBytesToBytesInternal(bytes, 0, bytes.Length, false));
+        }
+
+        private static byte[] UrlEncodeBytesToBytesInternal(byte[] bytes, int offset, int count, bool alwaysCreateReturnValue)
+        {
+            int cSpaces = 0;
+            int cUnsafe = 0;
+
+            // count them first
```
Comment (davidsh) [inline src/System.Net.WebClient/src/System/Net/WebClient.cs]: nit: Pls follow current convention for comments in networking code. Uppercase first word and end w/ period.

Comment (schen381) [inline src/System.Net.WebClient/src/System/Net/WebClient.cs]: ok

**Comment by davidsh** [comment]
Please run outerloop tests on this PR using commands like:

>@dotnet-bot test outerloop netcoreapp Windows_NT Debug
@dotnet-bot test outerloop netcoreapp Ubuntu16.04 Debug

Also, please make sure to verify this manually on 'netfx' configuration using something like:

>msbuild /t:rebuildandtest /p:TargetGroup=netfx /p:Outerloop=true

**Comment by schen381** [comment]
@davidsh 

>Also, please make sure to verify this manually on 'netfx' configuration using something like:

I have tested on my local machine

**Comment by schen381** [comment]
@dotnet-bot test outerloop netcoreapp Windows_NT Debug
@dotnet-bot test outerloop netcoreapp Ubuntu16.04 Debug

**Comment by schen381** [comment]
@dotnet-bot test outerloop netcoreapp Windows_NT Debug
@dotnet-bot test outerloop netcoreapp Ubuntu16.04 Debug
@dotnet-bot test Innerloop OSX10.12 Release x64 Build and Test

**Comment by davidsh** [comment]
@stephentoub PTAL

**Pull request merged**